### PR TITLE
Cscoiva 976

### PR DESCRIPTION
**Pull request by jpy-valamis**
Fixed PerustelutOpiskelijavuodet to determine the existence of changes from opiskelijavuodet instead of perustelut.
Made Difference have a read-only mode and expanded its stories.
Now using the read-only Difference as part of PerustelutOpiskelijavuodet (in Perustelut and Yhteenveto) view to show the applied differences in amounts.

### CODE REVIEW

File: src/components/02-organisms/Difference/index.js [hunk 1]
```diff
@@ -2,12 +2,15 @@ import React, { useState, useCallback, useEffect } from "react";
 import PropTypes from "prop-types";
 import TextField from "@material-ui/core/TextField";
 import Typography from "@material-ui/core/Typography";
+import * as R from "ramda";
```
Comment (sz1521) [inline src/components/02-organisms/Difference/index.js]: "R.path" only needed.

File: src/components/02-organisms/Difference/index.js [hunk 2]
```diff
@@ -78,7 +86,7 @@ const Difference = ({
         </div>
         <div className="flex-1 flex-col">
           <Typography>{inputAreaTitle}</Typography>
-          <div>
+          {!readonly && <div>
```
Comment (sz1521) [inline src/components/02-organisms/Difference/index.js]: Is this div really needed?

File: src/components/02-organisms/Difference/index.js [hunk 3]
```diff
@@ -87,7 +95,10 @@ const Difference = ({
               }
               value={value}
             />
-          </div>
+          </div>}
+          {readonly && <div>
```
Comment (sz1521) [inline src/components/02-organisms/Difference/index.js]: Should this div be span or p ?

File: src/services/lomakkeet/perustelut/muut.js [hunk 4]
```diff
@@ -598,13 +598,43 @@ export const getOppisopimusPerusteluLomake = isReadOnly => {
...
-  isReadOnly = false
+  isReadOnly = false,
+  changeObject,
+  differenceComponentTitles
 ) => {
   const checkboxes = getMuutostarveCheckboxes(checkboxItems, locale, isReadOnly);
   return [
+    {
+      anchor: "vahimmaisopiskelijavuodet",
+      title: "Haettava määrä",
```
Comment (sz1521) [inline src/services/lomakkeet/perustelut/muut.js]: Localization should be done.

File: src/services/lomakkeet/perustelut/muut.js [hunk 5]
```diff
@@ -632,8 +662,15 @@ export const getOpiskelijavuodetVahimmaisopiskelijavuosimaaraPerustelulomake = (
   ];
 };
 
-export const getOpiskelijavuodetVaativaKoulutusPerustelulomake = (isReadOnly = false) => {
+export const getOpiskelijavuodetVaativaKoulutusPerustelulomake = (isReadOnly = false, changeObject, differenceComponentTitles) => {
   return [
+    {
+      anchor: "vaativatuki",
+      title: "Haettava määrä",
```
Comment (sz1521) [inline src/services/lomakkeet/perustelut/muut.js]: Localization should be done.

File: src/services/lomakkeet/perustelut/muut.js [hunk 6]
```diff
@@ -653,8 +690,15 @@ export const getOpiskelijavuodetVaativaKoulutusPerustelulomake = (isReadOnly = f
   ];
 };
 
-export const getOpiskelijavuodetSisaoppilaitosmuotoinenOpetusPerustelulomake = (isReadOnly = false) => {
+export const getOpiskelijavuodetSisaoppilaitosmuotoinenOpetusPerustelulomake = (isReadOnly = false, changeObject, differenceComponentTitles) => {
   return [
+    {
+      anchor: "sisaoppilaitos",
+      title: "Haettava määrä",
```
Comment (sz1521) [inline src/services/lomakkeet/perustelut/muut.js]: Localization should be done.

File: src/services/lomakkeet/perustelut/muut.js (hunk 4 above)
Comment (jpy-valamis) [reply src/services/lomakkeet/perustelut/muut.js]: Not in scope of this issue. Localisation should be done for all matter in muut.js. Separate, bigger issue.

File: src/components/02-organisms/Difference/index.js (hunk 1 above)
Comment (jpy-valamis) [reply src/components/02-organisms/Difference/index.js]: R.path was the cleanest way of safely reaching a nested value from possibly undefined variable. I think it's justified as Ramda is used everywhere anyway.

File: src/components/02-organisms/Difference/index.js (hunk 2 above)
Comment (jpy-valamis) [reply src/components/02-organisms/Difference/index.js]: No. This was just following the pattern of how the component was done. I've removed all unnecessary divs now.

File: src/components/02-organisms/Difference/index.js (hunk 3 above)
Comment (jpy-valamis) [reply src/components/02-organisms/Difference/index.js]: Removed completely.

**Pull request merged**